dspy/clients/lm.py

```diff
@@ -110,6 +116,8 @@ def __call__(self, prompt=None, messages=None, **kwargs):
                 }
                 for c in response["choices"]
             ]
+        elif tools:
```

Comment:
I think the if statements here need to be refactored. Right now, you can only ask for logprobs _or_ tools, but not both.

I think we should move the if statement about logprobs and tools etc inside the loop over `choices`?

Reply:
good catch, fixed!